src/control/SConscript

```diff
@@ -66,9 +66,27 @@ def install_go_bin(denv, gosrc, gopath, libs, name, install_name):
     src = [mod_src]
     if libs is not None:
         src.extend(libs)
+    # generate a unique build id per binary for use by RPM
+    # https://fedoraproject.org/wiki/PackagingDrafts/Go#Build_ID
+    def gen_build_id():
+        from os import urandom
+        from binascii import b2a_hex
+        buildid = b2a_hex(urandom(20))
+        if isinstance(buildid, str):
+            # Python 2 is a str already, using decode() here transforms it
```

Comment:
Nit: Comment doesn't seem to match code anymore...

Reply:
Not sure I'm seeing the mismatch.  The comment is basically saying, "use it as is" on Python 2 which is what the code is doing.

Reply:
The comment in the python2 branch states:
"Python 2 is a str already, using decode() here transforms it into a 'unicode' type" ... But in the python2 branch, there's no use of decode. Was just something that jumped out at me as a reader. Something like "Python 2 is a str, no need to decode" would be clearer.

Reply:
Ahh.  I see the confusion.  If I were more verbose with my comment I guess I would have written:

_In Python 2 **buildid** is a str already; using decode() on it here transforms it into a 'unicode' type, so we're not doing that, like we do below for Python3 where we need to decode() it from a bytes into a str_